diff -r ef801a3bf79a extension.cpp
--- a/extension.cpp	Sat Jul 23 18:15:27 2011 +0100
+++ b/extension.cpp	Sat Aug 06 13:09:58 2011 +0200
@@ -24,6 +24,8 @@
 
 SMEXT_LINK(&g_UnlockBlock);
 
+IForward * g_pForwardItemHasBeenUpdated = NULL;
+
 ICvar *icvar = NULL;
 
 ConVar UnlockBlockVersion("unlockblock_version", SMEXT_CONF_VERSION, FCVAR_SPONLY|FCVAR_REPLICATED|FCVAR_NOTIFY, SMEXT_CONF_DESCRIPTION " Version");
@@ -74,6 +76,23 @@
 	CEconItemDefinition *pPersistentItemDefinition = GetItemDefinition(g_pPersistentItemSystem, pItem->m_iItemDefinitionIndex);
 	int slot = *(int *)((char *)pPersistentItemDefinition + ItemSlotOffset);
 
+	// Summon forward
+	cell_t cellResults = 0;
+	cell_t cellOverrideHandle = 0;
+	g_pForwardItemHasBeenUpdated->PushCell(pItem->m_iEntityQuality);
+	g_pForwardItemHasBeenUpdated->PushCell(pItem->m_iItemDefinitionIndex);
+	g_pForwardItemHasBeenUpdated->Execute(&cellResults);
+
+	switch(cellResults) {
+		case Pl_Changed:
+		case Pl_Handled:
+		case Pl_Stop:
+			{
+				pItem->m_iInventoryPosition &= ~0x0FFF0000;
+				break;
+			}
+	}
+
 	//META_CONPRINTF("%d = %d\n", pItem->m_iItemDefinitionIndex, slot);
 
 	if ((!ShowWeapons.GetBool() && (slot >= 0 && slot <= 6)) ||
@@ -145,6 +164,10 @@
 		return false;
 	}
 
+	// Create forwards
+	g_pForwardItemHasBeenUpdated = g_pForwards->CreateForward("UnlockBlock_ItemHasBeenUpdated", ET_Hook, 2, NULL, Param_Cell, Param_Cell);
+
+
 	return true;
 }
 
@@ -168,6 +191,7 @@
 void UnlockBlock::SDK_OnUnload() 
 {
 	gameconfs->CloseGameConfigFile(g_pGameConf);
+	g_pForwards->ReleaseForward(g_pForwardItemHasBeenUpdated);
 }
 
 bool UnlockBlock::SDK_OnMetamodUnload(char *error, size_t maxlen)
diff -r ef801a3bf79a extension.hpp
--- a/extension.hpp	Sat Jul 23 18:15:27 2011 +0100
+++ b/extension.hpp	Sat Aug 06 13:09:58 2011 +0200
@@ -109,5 +109,5 @@
 void ItemHasBeenUpdated(CTFPlayerInventory *thisptr, CEconItemView *pItem, bool a, bool b);
 CEconItemDefinition *GetItemDefinition(CEconItemSystem *thisptr, int iItemDefinitionIndex);
 CEconItemSystem *ItemSystem();
-
+extern IForward *g_pForwardItemHasBeenUpdated;
 #endif // _INCLUDE_SOURCEMOD_EXTENSION_PROPER_H_
diff -r ef801a3bf79a sdk/smsdk_config.hpp
--- a/sdk/smsdk_config.hpp	Sat Jul 23 18:15:27 2011 +0100
+++ b/sdk/smsdk_config.hpp	Sat Aug 06 13:09:58 2011 +0200
@@ -56,7 +56,7 @@
 #define SMEXT_CONF_METAMOD		
 
 /** Enable interfaces you want to use here by uncommenting lines */
-//#define SMEXT_ENABLE_FORWARDSYS
+#define SMEXT_ENABLE_FORWARDSYS
 //#define SMEXT_ENABLE_HANDLESYS
 //#define SMEXT_ENABLE_PLAYERHELPERS
 //#define SMEXT_ENABLE_DBMANAGER
